comp/otelcol/collector/impl/collector.go

```diff
@@ -81,8 +84,79 @@ func (c *converterFactory) Create(_ confmap.ConverterSettings) confmap.Converter
 	return c.converter
 }
 
+func newConfigProviderSettings(reqs Requires, enhanced bool) otelcol.ConfigProviderSettings {
+	converterFactories := []confmap.ConverterFactory{
+		expandconverter.NewFactory(),
+	}
+
+	if enhanced {
+		converterFactories = []confmap.ConverterFactory{
+			expandconverter.NewFactory(),
+			&converterFactory{converter: reqs.Provider},
+		}
+	}
+
+	return otelcol.ConfigProviderSettings{
+		ResolverSettings: confmap.ResolverSettings{
+			URIs: reqs.URIs,
+			ProviderFactories: []confmap.ProviderFactory{
+				fileprovider.NewFactory(),
+				envprovider.NewFactory(),
+				yamlprovider.NewFactory(),
+				httpprovider.NewFactory(),
+				httpsprovider.NewFactory(),
+			},
+			ConverterFactories: converterFactories,
+		},
+	}
+}
+
+func addFactories(reqs Requires, factories otelcol.Factories) {
+	if v, ok := reqs.LogsAgent.Get(); ok {
+		factories.Exporters[datadogexporter.Type] = datadogexporter.NewFactory(reqs.TraceAgent, reqs.Serializer, v, reqs.SourceProvider, reqs.StatsdClientWrapper)
+	} else {
+		factories.Exporters[datadogexporter.Type] = datadogexporter.NewFactory(reqs.TraceAgent, reqs.Serializer, nil, reqs.SourceProvider, reqs.StatsdClientWrapper)
+	}
+	factories.Processors[infraattributesprocessor.Type] = infraattributesprocessor.NewFactory(reqs.Tagger)
+	factories.Extensions[ddextension.Type] = ddextension.NewFactory(reqs.ConfigStore)
+}
+
+// getConfig returns the *otelcol.Config from the slice of URIs. If enhanced is
+// true, it returns the enhanced config, else it returns the provided config.
+func getConfig(reqs Requires, enhanced bool) (*otelcol.Config, error) {
+	ocp, err := otelcol.NewConfigProvider(newConfigProviderSettings(reqs, enhanced))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create configprovider: %w", err)
+	}
+
+	factories, err := reqs.CollectorContrib.OTelComponentFactories()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
can we move them to a different file/package? This is not part of collector implementation.

Reply:
Moved the logic into configstore in: https://github.com/DataDog/datadog-agent/pull/27337.